android-sdk/src/main/java/com/checkout/android_sdk/threeds/ThreedsWebView.kt

```diff
@@ -0,0 +1,50 @@
+package com.checkout.android_sdk.threeds
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.webkit.WebView
+import com.checkout.android_sdk.FramesLogger
+import com.checkout.android_sdk.FramesLogger.Companion.log
+import com.checkout.android_sdk.View.data.LoggingState
+
+@SuppressLint("ViewConstructor")
+internal class ThreedsWebView constructor(
+    context: Context,
+    val urlToLoad: String,
+    private val loggingState: LoggingState,
+    private val framesLogger: FramesLogger,
+) : WebView(context) {
+
+    init {
+        loadUrl(urlToLoad)
+        configureSettings()
+        if (visibility == VISIBLE) {
+            sendThreedsWebViewPresentedEvent()
+        }
```

Comment:
I don't think this is the correct place to call `sendThreedsWebViewPresentedEvent()` from. This would be called before the `ThreedsWebView` is added to the view. Should this instead be called from `PaymentForm.handle3DS()`?

Reply:
I have logged this event when the webview is initialised and moved it to the custom class. We can call it from handle 3ds method of paymentform. However, it does not have any return argument to verify by mocking this method and its public method for the external merchant. Additionally, we can simply remove the visibility check and call it in init method.

Reply:
Yeah I get that. Is there a lifecycle method we could call it from instead?

Reply:
We have webview and set up the theWebviewClient (with callbacks) to it. To verify presenting events of webview(created dynamically), we can verify view visibility by using (getVisibility method like in IOS viewdidLoad) in paymentform handle3DS class as well. but, it does have a dependency on frameslogger to unit test.  So, I have moved this event to the class in the initialisation method of it.